client/sites/hooks/test/use-transfer-with-software-status-query.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { renderHook, waitFor } from '@testing-library/react';
+import nock from 'nock';
+import React from 'react';
+import { useTransferWithSoftwareStatus } from '../use-transfer-with-software-status-query';
+
+// Mock wpcom
+jest.mock( '@automattic/calypso-config', () => ( {
+	isEnabled: jest.fn(),
+} ) );
+
+const mockSuccessResponse = {
+	software_transfer_status: 'pending',
+};
+
+describe( 'useTransferWithSoftwareStatus', () => {
+	beforeAll( () => nock.disableNetConnect() );
+
+	beforeEach( () => nock.cleanAll() );
+
+	afterEach( () => jest.resetAllMocks() );
+
+	it( 'should fetch transfer status successfully', async () => {
+		const queryClient = new QueryClient();
+		const wrapper = ( { children }: { children: React.ReactNode } ) => (
+			<QueryClientProvider client={ queryClient }>{ children }</QueryClientProvider>
+		);
+		nock( 'https://public-api.wordpress.com' )
+			.get( '/wpcom/v2/sites/123/atomic/transfer-with-software/456' )
+			.query( { http_envelope: 1 } )
+			.reply( 200, mockSuccessResponse );
+
+		const { result } = renderHook( () => useTransferWithSoftwareStatus( 123, 456 ), { wrapper } );
+
+		await waitFor( () => expect( result.current.isSuccess ).toBe( true ) );
+		expect( result.current.data ).toEqual( mockSuccessResponse );
```

Comment:
Can you note why we have one check in the `waitFor()`  but not the other? It looks weird to me to have one check wrapped and not the other.

Reply:
As a further consideration, do we want to add a test that ensures we ignore additional fields in the response?

Reply:
Oops, pretty sure I was messing around trying to get the test to work and moved it out of the `waitFor` unintentionally. 🙃 And yep, we can add a test that ignores the additional fields. I'm not sure if we're going to end up using the additional fields but we can always update the tests and the hook again if we need them.